lib/recursive-methods.rb

```diff
@@ -1,49 +1,90 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the size of the number input
+# Space complexity: O(n) where n is the size of the number input
 def factorial(n)
-    raise NotImplementedError, "Method not implemented"
+  if n < 0
+    raise ArgumentError
+  elsif n == 0 || n == 1
+    return 1
+  else
+    return n * (factorial(n - 1))
+  end
 end
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the length of the string
+# Space complexity: O(n) where n is the length of the string
 def reverse(s)
-    raise NotImplementedError, "Method not implemented"
+  if s.length <= 1
+    return s
+  else
+    return rev_str = reverse(s[1..-1]) + s[0]
+  end
 end
 
 # Time complexity: ?
 # Space complexity: ?
 def reverse_inplace(s)
-    raise NotImplementedError, "Method not implemented"
```

Comment:
Think about making a helper which takes the string, a left index and right index.  The base case is when `left_index >= right_index`  Each recursive call you swap the characters at `left_index` or `right_index` and then make a recursive call with the string, and `left_index + 1` and `right_index -1`.